pkg/upgrade/1.0.0-1.1.0/v1alpha1/csp_upgrade.go

```diff
@@ -252,6 +266,17 @@ func (c *cstorCSPOptions) poolInstanceUpgarde(openebsNamespace string) error {
 		return err
 	}
 
+	err = c.waitForCurrentVersion()
+	if err != nil {
+		statusObj.Message = "failed to verify versiondetails for cstor pool"
+		statusObj.Reason = strings.Replace(err.Error(), ":", "", -1)
+		c.utaskObj, uerr = updateUpgradeDetailedStatus(c.utaskObj, statusObj, openebsNamespace)
+		if uerr != nil && isENVPresent {
+			return uerr
+		}
+		return err
+	}
+
 	err = patchCSP(c.cspObj)
```

Comment:
don't we need to wait for `current` to become `desired`?

Reply:
added the wait function